test/parallel/test-beforeexit-event.js

```diff
@@ -31,12 +36,15 @@ function tryListen() {
   net.createServer()
     .listen(0)
     .on('listening', function() {
-      revivals++;
+      beforeProcess();
       this.close();
+    })
+    .on('error', (e) => {
+      throw e;
     });
 }
 
-process.on('exit', function() {
-  assert.equal(4, deaths);
-  assert.equal(3, revivals);
+process.on('exit', () => {
+  common.mustCall(beforeExit, 4);
+  common.mustCall(beforeProcess, 3);
```

Comment:
That's not how this works. :)

You will need to wrap the callback that you want to call *before* using it (as in, the variables at lines 9-15 already need the wrappers). Then the `exit` event listener can be removed.